sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -308,7 +322,7 @@ private void readIntBatch(int rowId, int num, ColumnVector column) throws IOExce
 
   private void readLongBatch(int rowId, int num, ColumnVector column) throws IOException {
     // This is where we implement support for the valid type conversions.
-    if (column.dataType() == DataTypes.LongType ||
+    if (column.dataType() == DataTypes.LongType || column.dataType() == DataTypes.TimestampType ||
         DecimalType.is64BitDecimalType(column.dataType())) {
       defColumn.readLongs(
```

Comment:
hmm. How does this work? readLongs is expecting to read parquet int64 physical types. How is it able to read this other physical type?